src/releaseNotesProvider.ts

```diff
@@ -0,0 +1,180 @@
+import { env, ExtensionContext } from "vscode";
+import * as vscode from "vscode";
+import * as semver from "semver";
+import { Remarkable } from "remarkable";
+import http from "https";
+
+const versionKey = "metals-server-version";
+
+// prettier-ignore
+const releaseNotesMarkdownUrl: Record<string, string> = {
+  "0.11.5": "https://raw.githubusercontent.com/scalameta/metals/main/website/blog/2022-04-28-aluminium.md",
+  "0.11.6": "https://raw.githubusercontent.com/scalameta/metals/main/website/blog/2022-06-03-aluminium.md",
+};
+
+/**
+ * Show release notes if possible, swallow errors since its not a crucial feature.
+ * Treats snapshot versions like 0.11.6+67-926ec9a3-SNAPSHOT as a 0.11.6.
+ */
+export async function showReleaseNotesIfNeeded(
+  context: ExtensionContext,
+  serverVersion: string,
+  outputChannel: vscode.OutputChannel
+) {
+  try {
+    await showReleaseNotes(context, serverVersion);
+  } catch (error) {
+    outputChannel.appendLine(
+      `Couldn't show release notes for Metals ${serverVersion}`
+    );
+    outputChannel.appendLine(`${error}`);
+  }
+}
+
+async function showReleaseNotes(
+  context: ExtensionContext,
+  currentVersion: string
+): Promise<void> {
+  const state = context.globalState;
+
+  const version = getVersion();
+  const releaseNotesUrl = version
+    ? releaseNotesMarkdownUrl[version]
+    : undefined;
+  if (isNotRemote() && version && releaseNotesUrl) {
+    await showPanel(version, releaseNotesUrl);
+  }
+
+  async function showPanel(version: string, releaseNotesUrl: string) {
+    const releaseNotes = await getReleaseNotesMarkdown(releaseNotesUrl);
+
+    if (typeof releaseNotes === "string") {
+      const panel = vscode.window.createWebviewPanel(
+        `scalameta.metals.whatsNew`,
+        `Metals ${version} release notes`,
+        vscode.ViewColumn.One
+      );
+
+      panel.webview.html = releaseNotes;
+      panel.reveal();
+
+      // set wrong value for local development
+      // const newVersion = "0.11.5";
+
+      const newVersion = version;
```

Comment:
I left this on purpose for local development/testing reasons, just swap `//` betwen lines to always get release notes or to overwrite saved version.

Reply:
actually it doesn't work because after version is set execution won't get here :c